tests/ui/borrowck/suggest-as-ref-on-mut-closure.stderr

```diff
@@ -5,11 +5,14 @@ LL |     cb.map(|cb| cb());
    |     ^^^--------------
    |     |  |
    |     |  `*cb` moved due to this method call
-   |     help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
    |     move occurs because `*cb` has type `Option<&mut dyn FnMut()>`, which does not implement the `Copy` trait
    |
 note: `Option::<T>::map` takes ownership of the receiver `self`, which moves `*cb`
   --> $SRC_DIR/core/src/option.rs:LL:COL
+help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
+   |
+LL |     cb.map(|cb| cb()).as_ref();
```

Comment:
perhaps this was the intention? @estebank

Reply:
I think the intent was for `cb.as_ref().map(|cb| cb());`.

Reply:
I mean the intention why it wasn't the proper thing in the first place. since this is only naughty one where other 2 updated `.stderr` [works as expected.](https://github.com/rust-lang/rust/pull/104055/commits/4dd47c213635d1bc15ed0dfb87e8a7772453f4bd#diff-15a9c3229cae68c0e43e4c9a2f807ace499822e1f7acd6d435e5ec3feab67018R36)

I tweaked with `eager_subdiag` and `span.other_method()` yet won't work.

Reply:
It was intentional, see https://github.com/rust-lang/rust/commit/e37565d2db8d7cf92a6d1df535343d4b2881fc35